### PR TITLE
✨ [Feature] #102: In filter modal creator dropdown should be a card list with image

### LINKED ISSUE #102: ✨ [Feature request]: In filter modal creator dropdown should be a card list with image
**Is your feature request related to a problem? Please describe.**
Currently, the creator dropdown is a flat list. It doesn't give enough context of the GitHub profile

**Describe the solution you'd like**
Use a contact card drop-down  instead

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- I worked on adding support for creator images in the creators select dropdown.
- For Consistency, I have changed the other select inputs to `@mui/material` `<Select/>` Component.
- Added fallback option if avatar URL is not present.

Fixes #102 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

image

image

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Sachin-chaurasiya** [comment]
@atapas @koustov, PR is ready for review.

**Comment by 6km** [comment]
@Sachin-chaurasiya Looks good 🔥🔥

**Comment by Sachin-chaurasiya** [comment]
> @Sachin-chaurasiya Looks good 🔥🔥

Thanks @6km  🚀

**Comment by koustov** [comment]
I see this branch has not been deployed for preview. Will see the reason and then initiate my review

**Comment by Sachin-chaurasiya** [comment]
@koustov @atapas, I have updated the PR with main, Please review.

**Review by atapas — DISMISSED** [review]
LGTM

**Comment by atapas** [comment]
@koustov please take a look

**Comment by koustov** [comment]
I don't see the preview deployment. It says 404. @atapas can you check

**Comment by atapas** [comment]
> I don't see the preview deployment. It says 404. @atapas can you check

As we moved the base in vervet, old links won’t work. Please trigger a build manually.

**Comment by koustov** [comment]
> > I don't see the preview deployment. It says 404. @atapas can you check
> 
> As we moved the base in vervet, old links won’t work. Please trigger a build manually.

I am not able to pull the build on vercel. Still need to see from API side. Meanwhile, @Sachin-chaurasiya can you make a dummy check-in ( a very tiny change something like adding an empty space) so that vercel build gets triggered?

**Comment by Sachin-chaurasiya** [comment]
> > > I don't see the preview deployment. It says 404. @atapas can you check
> > 
> > 
> > As we moved the base in vervet, old links won’t work. Please trigger a build manually.
> 
> I am not able to pull the build on vercel. Still need to see from API side. Meanwhile, @Sachin-chaurasiya can you make a dummy check-in ( a very tiny change something like adding an empty space) so that vercel build gets triggered?

Hey @koustov , Done.

File: package.json [hunk 1]
```diff
@@ -32,9 +32,10 @@
     "react-redux": "^8.0.1",
     "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.0",
+    "react-select": "^5.4.0",
```
Comment (koustov) [inline package.json:35]: As we already have MUI, can you try customizing `autocomplete` or `select` to achieve the dropdown with avatar image?

**Review by koustov — CHANGES_REQUESTED** [review]
I still have the vercel issue. Until we fix it, I have dropped one comment. Check if you can address it.

File: package.json (hunk 1 above)
Comment (Sachin-chaurasiya) [reply package.json:35]: Sure @koustov, I will have a look.

Comment (Sachin-chaurasiya) [reply package.json:35]: @koustov, I checked the MUI Select component and it does provide support for showing images in options, but it does not support displaying images for selected options.
Something like this.
image

Comment (koustov) [reply package.json:35]: @Sachin-chaurasiya I just made a quick and dirty one for reference. https://codesandbox.io/s/stoic-ishizaka-3nbwiu

You have to make tons of changes on top of it however it satisfies what we are looking for.

Comment (Sachin-chaurasiya) [reply package.json:35]: Thanks, @koustov, I will work on it.

Comment (Sachin-chaurasiya) [reply package.json:35]: Hey @koustov, I found a better way of doing it, we can use the `renderValue` prop to render the creator with an image and label.

You can checkout the implementation here https://codesandbox.io/s/custom-select-wfy40v?file=/demo.tsx

**Comment by Sachin-chaurasiya** [comment]
@koustov @atapas, I have removed the `react-select` and used the material UI select component instead.

**Comment by Sachin-chaurasiya** [comment]
Closing this PR because every time it needs to be authorized for deployment.
Created new PR https://github.com/reactplay/react-play/pull/483

**Pull request closed without merging**